packages/arb-token-bridge-ui/src/hooks/useNetworksAndSigners.tsx

```diff
@@ -200,10 +187,15 @@ export function NetworksAndSignersProvider(
         .catch(() => {
           getL2Network(web3Provider)
             .then(async l2Network => {
+              if(l2Network.chainID != _selectedL2ChainId){
+                setResult({ status: UseNetworksAndSignersStatus.NOT_SUPPORTED })
+                return
+              }
```

Comment:
I think we should do this check earlier in the flow (i think other stuff would break before hitting this here like `         const l2Provider = new JsonRpcProvider(rpcURLs[l2NetworkChainId])`